src/domains/projects/projects.handlers.ts

```diff
@@ -0,0 +1,103 @@
+import type { z } from '@hono/zod-openapi';
+
+import { eq } from 'drizzle-orm';
+
+import type { insertProjectsSchema, patchProjectsSchema, selectProjectsSchema } from '@/db/schema';
+import type { Result } from '@/lib/types';
+
+import { db } from '@/db';
+import { projects } from '@/db/schema';
+
+export type Project = z.infer<typeof selectProjectsSchema>;
+export type CreateProjectInput = z.infer<typeof insertProjectsSchema>;
+export type UpdateProjectInput = z.infer<typeof patchProjectsSchema>;
+
+export async function getAllProjects(): Promise<Result<Project[]>> {
+  try {
+    const projectList = await db.select().from(projects);
+    return { ok: true, data: projectList as Project[] };
```

Comment:
After making the `schema.ts` change for the Json `metadata` column we no longer need this typecast.
```suggestion
    return { ok: true, data: projectList };
```